lib/icinga/downtime.cpp

```diff
@@ -317,9 +317,12 @@ void Downtime::RemoveDowntime(const String& id, bool cancelled, bool expired, co
 	String config_owner = downtime->GetConfigOwner();
 
 	if (!config_owner.IsEmpty() && !expired) {
-		Log(LogWarning, "Downtime")
-			<< "Cannot remove downtime '" << downtime->GetName() << "'. It is owned by scheduled downtime object '" << config_owner << "'";
-		return;
+		String errorMessage = "Cannot remove downtime '" + downtime->GetName() +
+				"'. It is owned by scheduled downtime object '" + config_owner + "'";
+
+		Log(LogWarning, "Downtime") << errorMessage;
```

Comment:
When `RemoveDowntime` is called from `ExternalCommandProcessor`, this is now logged twice. I'd argue for removing this log message here, as with this PR, the failure will properly be reported to the entity performing the API request. I think this is more helpful than a log message where you don't really know which request caused it. But I could also understand if we want to keep some log message in this case. @Al2Klimov @N-o-X Someone having a second opinion here?

Reply:
I removed it from ``downtime.cpp`` but instead I logged the error message in ``ApiActions``.